bench/workgen/workgen.cpp

```diff
@@ -465,40 +507,27 @@ WorkloadRunner::start_tables_create(WT_CONNECTION *conn)
             continue;
         }
 
-        retries = 0;
-        creates = 0;
-        while (creates < _workload->options.create_count) {
+        // Add app_metadata to the config to indicate the table was created dynamically which means
+        // it can be selected for random deletion. We want this information to persist over restart.
+        const std::string config(
+          "key_format=S,value_format=S,app_metadata=\"" + DYN_TABLE_APP_METADATA);
 
+        int creates = 0, retries = 0;
+        while (
+          !stopping && creates < _workload->options.create_count && retries < TABLE_MAX_RETRIES) {
             // Generate a table name from the user specified prefix and a random alphanumeric
             // sequence.
+            char rand_chars[DYNAMIC_TABLE_LEN];
             gen_random_table_name(rand_chars, _rand_state);
-            uri = "table:";
-            uri += _workload->options.create_prefix;
-            uri += rand_chars;
-
-            if (_workload->options.mirror_tables) {
-                // The mirror table name is the table name with the user specified suffix.
-                mirror_uri = uri + _workload->options.mirror_suffix;
-                config = base_config + mirror_uri + "," + BASE_TABLE_APP_METADATA + "true\"";
-            }
+            const std::string uri("table:" + _workload->options.create_prefix + rand_chars);
 
-            // Create the table. Simply continue on failure.
-            if (create_table(session, config, uri, mirror_uri, true) == 0) {
-                VERBOSE(*_workload, "Created base table '" << uri << "'");
-                if (_workload->options.mirror_tables) {
-                    // Create the mirror. Retry on failure and throw an exception after
-                    // making too many retry attempts.
-                    config = base_config + uri + "," + BASE_TABLE_APP_METADATA + "false\"";
-                    do
-                        status = create_table(session, config, mirror_uri, uri, false);
-                    while (status == EBUSY && ++retries < TABLE_MAX_RETRIES);
-                    if (status != 0)
-                        THROW_ERRNO(
-                          status, "Failed to create mirror table '" << mirror_uri << "'.");
-                    VERBOSE(*_workload, "Created mirror table '" << mirror_uri << "'");
-                }
+            // Create the table and its mirror if enabled.
+            int ret = create_table(session, config, uri, _workload->options.mirror_tables);
```

Comment:
The earlier logic ignored the failure in creating a base table, and just continued on- but retried creating a mirror in case we get EBUSY (not EEXIST). Now we retry creating both, including when failing with EEXIST. I forgot the reason why we coded that way - maybe because there was a chance to randomly generate a table name which just got deleted?

Anyways, I think we can move to a different logic, of ignoring both EBUSY and EEXIST and moving on, now that we create a mirror first. But raising an exception and asserting if the mirror gets created, but base creation fails. I think you are doing the error-handling part for base creation already.

Reply:
I see, I missed a part indeed.
See 5d99c897581ab832a14e7bdbca511c68a61ab73a where we are now retrying to create the base when mirroring is enabled up to `TABLE_MAX_RETRIES` times.